app/Svg/Parser.hs

```diff
@@ -586,6 +585,24 @@ updateShape fill r =
 addTuples :: Point -> Point -> Point
 addTuples (a,b) (c,d) = (a + c, b + d)
 
+-- | Apply a matrix transformation to a point.
+matrixPointMultiply :: Matrix -> Point -> Point
+matrixPointMultiply m p =
+    let [x, y, _] = concat $ matrixMultiply m (pointToMatrix p)
```

Comment:
The `concat` isn't necessary, you can just do more precise pattern-matching (`[[x, y, _]]`).